api/hasura/actions/_handlers/createUsers.ts

```diff
@@ -135,6 +137,162 @@ async function handler(req: VercelRequest, res: VercelResponse) {
     return opts;
   }, {} as { [aliasKey: string]: ValueTypes['mutation_root'] });
 
+  //check if names are used by other coordinape users
+  const { profiles: existingNames } = await adminClient.query(
+    {
+      profiles: [
+        {
+          where: {
+            _or: newUsers.map(user => {
+              return {
+                _and: [
+                  { name: { _eq: user.name } },
+                  { address: { _nilike: user.address } },
+                ],
+              };
+            }),
+          },
+        },
+        { name: true },
+      ],
+    },
+    {
+      operationName: 'createUsers_getExistingNames',
+    }
+  );
+
+  if (existingNames.length > 0) {
+    const names = existingNames.map(u => u.name);
+    return errorResponseWithStatusCode(
+      res,
+      {
+        message: `Users list contains ${
+          names.length > 1
+            ? 'names used by other coordinape users'
+            : 'a name used by another coordinape user'
+        }: ${names}`,
+      },
+      422
+    );
+  }
+
+  const { profiles } = await adminClient.query(
```

Comment:
You probably can save on the query call if you merge the create & update profile mutation into an upsert like this. Think this should work the same.

```
import { profiles_constraint, profiles_update_column } from '../../../../api-lib/gql/__generated__/zeus';

const createProfilesMutations = adminClient.mutate({
    insert_profiles:[
      {
        objects: newUsers.map( user => {
          return {
            address: user.address.toLowerCase(),
            name: user.name,
          }
        }),
        on_conflict: {
          constraint: profiles_constraint.profiles_address_key,
          update_columns: [profiles_update_column.name],
          where: {
            name: {_is_null: true}
          }
        }
      },
      {
        returning: {
          id: true,
        }
      }
    ]
  },
  {
    operationName: 'createUsers_createProfiles',
  });
  ```